Automattic/woocommerce-payments
# Performance improvements for disputes reminder task list

Fixes #9716 

#### Changes proposed in this Pull Request

Performance improvements for the WC Admin Disputes Reminder Task.

Previously this feature could cause some errored accounts (and some non-erroring accounts) to make excessive calls to the Server's Disputes API. Due to the way that the API requests were cached, this only affected a small number of merchants.

We believe this occurred when the Account was Rejected or Suspended, the call to `GET disputes` would error preventing caching.  

Improvements:
 - Task is only loaded when 
   - logged-in user has `manage_woocommerce` permission
   - Account is not rejected or suspended
 - API requests are delayed until the WC task list is initialized.

#### Additional Context

pdjTHR-4rD-p2

#### Testing instructions

1. Ensure your store has disputes expiring within 1 week.
2. Visit WC Dashboard, and confirm the Task Item is visible

Additional Notes:
 - If your store does not have disputes expiring within one week, you can modify your server database and adjust the due date of disputes. 
 - You may also need to clear the cache in your store - `DELETE FROM wp_options WHERE option_name='wcpay_active_dispute_cache';`
 - You can test an "errored" cache by altering the `wcpay_active_dispute_cache` option to end in `"errored";b:1;}` (ie. errored=true)
 - You can test an "expired" cache by altering the `fetched` value in `wcpay_active_dispute_cache`


-------------------


**Post merge**
